### PR TITLE
make rom models trainable in preparation for learning ROM CSP constraints

### PR DESCRIPTION
**Pull request by tomsilver**

### CODE REVIEW

File: src/multitask_personalization/rom/models.py [hunk 1]
```diff
@@ -272,27 +312,25 @@ def __init__(
...
-        return self._reachable_points
+    def get_trainable_parameters(self) -> Any:
+        return self._rom_model_context_parameters.copy()
 
-    def update_parameters(self, parameters: NDArray) -> None:
-        """Update the ROM model parameters."""
-        self._rom_model_context_parameters = parameters
+    def set_trainable_parameters(self, params: Any) -> None:
+        assert isinstance(params, np.ndarray)
+        self._rom_model_context_parameters = params
```
Comment (RealZiangLiu) [inline src/multitask_personalization/rom/models.py:333]: Is making a copy of params needed here?

File: tests/rom/test_learned_rom.py [hunk 2]
```diff
@@ -15,20 +15,18 @@ def test_learned_rom_model():
...
-        == learned_rom_model.get_parameter_size()
-    )
-    # Test update_parameters is updating the parameters and reachable_points
-    parameters = learned_rom_model.get_rom_model_context_parameters()
-    # add small gaussian noise to parameters
-    n_prev_reachable_points = len(learned_rom_model.get_reachable_points())
+    parameters = learned_rom_model.get_trainable_parameters()
+    assert isinstance(parameters, np.ndarray)
+    assert parameters.shape == (4,)
+    # Add small gaussian noise to parameters
```
Comment (RealZiangLiu) [inline tests/rom/test_learned_rom.py:24]: misplaced comment, should go below

File: tests/rom/test_spherical_rom.py [hunk 3]
```diff
@@ -45,3 +45,17 @@ def test_spherical_rom_model():
         point = spherical_rom_model.sample_reachable_position(rng)
         distance = np.linalg.norm(point - sphere_center)
         assert distance < sphere_radius + 1e-6
+
+    # Test training the parameters.
+    init_params = spherical_rom_model.get_trainable_parameters()
+    assert np.isclose(init_params, sphere_radius)
+
+    data = [
+        (sphere_center + np.array([0.0, 0.0, 0.9]), False),
```
Comment (RealZiangLiu) [inline tests/rom/test_spherical_rom.py:54]: Is the label here flipped? 0.9 should be positive -> True, 1.1 negative -> False

File: src/multitask_personalization/rom/models.py (hunk 1 above)
Comment (tomsilver) [reply src/multitask_personalization/rom/models.py:333]: oh wait sorry I misread your comment. yes good idea!

**Pull request merged**